core/runtime/bindings.rs

```diff
@@ -73,43 +80,86 @@ pub(crate) fn create_external_references(
     });
   }
 
-  for ctx in ops {
-    if ctx.metrics_enabled() {
-      let ctx_ptr = ctx as *const OpCtx as _;
-      references.push(v8::ExternalReference { pointer: ctx_ptr });
-      references.push(v8::ExternalReference {
-        function: ctx.decl.slow_fn_with_metrics,
-      });
-      if let Some(fast_fn) = &ctx.decl.fast_fn_with_metrics {
-        references.push(v8::ExternalReference {
-          pointer: fast_fn.function as _,
-        });
-        references.push(v8::ExternalReference {
-          pointer: ctx.fast_fn_c_info.unwrap().as_ptr() as _,
-        });
-      }
-    } else {
-      let ctx_ptr = ctx as *const OpCtx as _;
-      references.push(v8::ExternalReference { pointer: ctx_ptr });
-      references.push(v8::ExternalReference {
-        function: ctx.decl.slow_fn,
-      });
-      if let Some(fast_fn) = &ctx.decl.fast_fn {
-        references.push(v8::ExternalReference {
-          pointer: fast_fn.function as _,
-        });
-        references.push(v8::ExternalReference {
-          pointer: ctx.fast_fn_c_info.unwrap().as_ptr() as _,
-        });
-      }
-    }
+  references.extend_from_slice(additional_references);
+
+  for ctx in &ops[..ops_in_snapshot] {
+    references.extend_from_slice(&ctx.external_references());
   }
 
-  references.extend_from_slice(additional_references);
+  for source in &sources[..sources_in_snapshot] {
+    references.push(v8::ExternalReference {
+      pointer: source as *const _ as _,
+    })
+  }
+
+  for ctx in &ops[ops_in_snapshot..] {
+    references.extend_from_slice(&ctx.external_references());
+  }
 
-  let refs = v8::ExternalReferences::new(&references);
-  let refs: &'static v8::ExternalReferences = Box::leak(Box::new(refs));
-  refs
+  for source in &sources[sources_in_snapshot..] {
+    references.push(v8::ExternalReference {
+      pointer: source as *const _ as _,
+    })
+  }
+
+  v8::ExternalReferences::new(&references)
+}
+
+/// Combine the snapshotted sources (which may be empty) with the loaded sources, and ensure that
```

Comment:
@bartlomieju Added some explanatory comments to this method.

Reply:
Thanks!